lib/utils/common.py

```diff
@@ -1740,70 +1780,87 @@ def _format_ns_stop_writes_metrics(
-            config = "max-used-pct"
-            usage: str | None = stats.get(metric, None)
-            bytes_total: str | None = stats.get("device_total_bytes", None)
-            threshold: str | None = stats.get(config, None)
-
-            if usage is None:
-                metric = "pmem_used_bytes"
-                usage = stats.get(metric, None)
-                bytes_total = stats.get("pmem_total_bytes", None)
+            metric, usage = _get_first_value_from_dict_with_key(
+                stats,
+                ("data_used_bytes", "device_used_bytes", "pmem_used_bytes"),
+                default_value=None,
+                return_type=int,
+            )
+            config, threshold = _get_first_value_from_dict_with_key(
+                stats,
+                ("storage-engine.stop-writes-used-pct", "storage-engine.max-used-pct"),
+                default_value=None,
+                return_type=int,
+            )
+            bytes_total: int | float | None = util.get_value_from_dict(
+                stats,
+                ("data_total_bytes", "device_total_bytes", "pmem_total_bytes"),
+                None,
+                return_type=int,
+            )
 
             if usage is not None and threshold is not None and bytes_total is not None:
-                use = int(usage)
-                thresh = int(bytes_total) * (int(threshold) / 100)
-                sw = _is_stop_writes_cause(use, thresh, stop_writes)
+                threshold = bytes_total * (threshold / 100)
+                sw = _is_stop_writes_cause(usage, threshold, stop_writes)
                 _create_stop_writes_entry(
                     stop_writes_metrics[node],
                     metric,
-                    use,
+                    usage,
                     sw,
-                    thresh,
+                    threshold,
                     config=config,
                     namespace=ns,
                 )
 
             metric = "memory_used_bytes"
             config = "stop-writes-pct"
-            usage: str | None = stats.get(metric, None)
-            bytes_total: str | None = stats.get("memory-size", None)
-            threshold: str | None = stats.get(config, None)
+            usage = util.get_value_from_dict(
+                stats, metric, default_value=None, return_type=int
+            )
+            bytes_total = util.get_value_from_dict(
+                stats, "memory-size", default_value=None, return_type=int
+            )
+            threshold = util.get_value_from_dict(
+                stats, config, default_value=None, return_type=int
+            )
 
             if usage is not None and threshold is not None and bytes_total is not None:
-                use = int(usage)
-                thresh = int(bytes_total) * (int(threshold) / 100)
-                sw = _is_stop_writes_cause(use, thresh, stop_writes)
+                threshold = int(bytes_total) * (int(threshold) / 100)
```

Comment:
Are the `int()` casts still needed? They are removed on line 1831

Reply:
nope, good catch